### DISCUSSION TITLE
API Results Data

### DISCUSSION THREAD

**Discussion by josh99smith** [Budibase/budibase#7573] [Help]
I am making API calls to a Sharepoint site through the Microsfot Graph API. I can not figure out how to get the results back as the fields only. I seem to be getting a bunch of other uneeded metadata. Below is one of the items being returned in the data. However, I only want the "fields" data to be returned so that I can populate tables with this data.

I am using a transformer, but I can't seem to get it to return the results I need.

`return data.value.fields` - This is returning no results

Here is the data that is returning with `return data.value` 

`{
  "@odata.etag": "\"4cca94cb-4ee5-412c-a1c1-e8c9edc4efe9,1\"",
  "createdDateTime": "2022-03-15T15:43:02Z",
  "eTag": "\"4cca94cb-4ee5-412c-a1c1-e8c9edc4efe9,1\"",
  "id": "43591",
  "lastModifiedDateTime": "2022-03-15T15:43:02Z",
  "webUrl": "https://odbtn.sharepoint.com/Our%20Daily%20Bread%20Staff%20Portal/Lists/Master%20Provider%20List/43591_.000",
  "createdBy": {
    "user": {
      "email": "jsmith@odbcacfp.org",
      "id": "aecd3b8c-1176-4f3a-bf5f-84cfe59c98e9",
      "displayName": "Joshua Smith"
    }
  },
  "lastModifiedBy": {
    "user": {
      "email": "jsmith@odbcacfp.org",
      "id": "aecd3b8c-1176-4f3a-bf5f-84cfe59c98e9",
      "displayName": "Joshua Smith"
    }
  },
  "parentReference": {
    "id": "dfc31b50-f85a-4243-807e-fd85790d5d28",
    "siteId": "odbtn.sharepoint.com,58545f71-7ceb-4ebb-80b2-5c60080a5db8,ec6564aa-3fc7-4d6d-99c1-5cd78af25227"
  },
  "contentType": {
    "id": "0x0100566BC883870CD0439E768FE9D5D7E1E9",
    "name": "Item"
  },
  "fields@odata.context": "https://graph.microsoft.com/v1.0/$metadata#sites('odbtn.sharepoint.com%2C58545f71-7ceb-4ebb-80b2-5c60080a5db8%2Cec6564aa-3fc7-4d6d-99c1-5cd78af25227')/lists('8f730b0e-2bf8-425a-ac59-e90cd51e18a8')/items('43591')/fields/$entity",
  "fields": {
    "@odata.etag": "\"4cca94cb-4ee5-412c-a1c1-e8c9edc4efe9,1\"",
    "Title": "Kidd's World",
    "Contact_x0020_Name": ",NA",
    "Phone_x0020_Number": "(423) 624-3669",
    "License_x0020_Type": "Center",
    "CACFP_x0020_Original_x0020_Start": "2016-11-01T04:00:00Z",
    "Address": "1905 Tunnel Blvd",
    "City": "Chattanooga",
    "State": "TN",
    "Monitor": "Broherd, Kasie",
    "LinkTitleNoMenu": "Kidd's World",
    "LinkTitle": "Kidd's World",
    "SiteTitle": "Kidd's World",
    "ProviderID": "C-404-RM-H",
    "Claims_x0020_ProccesorLookupId": "74",
    "Status": "Removed",
    "id": "43591",
    "ContentType": "Item",
    "Modified": "2022-03-15T15:43:02Z",
    "Created": "2022-03-15T15:43:02Z",
    "AuthorLookupId": "13",
    "EditorLookupId": "13",
    "_UIVersionString": "1.0",
    "Attachments": false,
    "Edit": "",
    "ItemChildCount": "0",
    "FolderChildCount": "0",
    "_ComplianceFlags": "",
    "_ComplianceTag": "",
    "_ComplianceTagWrittenTime": "",
    "_ComplianceTagUserId": ""
  }
}`

**Comment by shogunpurple** [comment]
@josh99smith according to your JSON, it should just be:

`return data.fields`

Working app export:
[JSON App-export-1662054031563.txt](https://github.com/Budibase/budibase/files/9472106/JSON.App-export-1662054031563.txt)

**Reply by josh99smith** [reply]
> @josh99smith according to your JSON, it should just be:
> 
> `return data.fields`
> 
> Working app export: [JSON App-export-1662054031563.txt](https://github.com/Budibase/budibase/files/9472106/JSON.App-export-1662054031563.txt)

Thanks a ton for your response! I tried `return data.fields`, however this does not return any data for me :(
